Recursive logit model

pabloguarda opened the pull request.

Commit:
Create .gitignore

Commit:
Update .gitignore

Commit:
Copy graph.py

Commit:
Copy paths, settings and types.py to have consistency between implementations

Commit:
Full implementation probabilistic a-star with logit recursion

Commit:
Last individual commit Pablo

Commit:
Update prod.txt

mbattifarano commented on examples/PG_console_testing.py:
```diff
@@ -0,0 +1,231 @@
+# =============================================================================
+
+# Internal modules
+from src.inverse_astar_search.network import *
```
`from inverse_astar_search.network [...]`

mbattifarano commented on requirements/dev.txt:
```diff
@@ -1 +1,4 @@
-pytest==5.4.1
+# pip install -r requirements/dev.txt
```
remove

mbattifarano commented on setup.py:
```diff
@@ -6,7 +6,7 @@ def read_requirements(filename):
 
 setup(
-    name="inverse-astar-search",
+    name="inverse-pastar.py-search",
```
revert

mbattifarano commented on src/inverse_astar_search/pastar.py:
```diff
@@ -0,0 +1,157 @@
+    #
+    # nx.set_edge_attributes(G, len_edges, attribute_label)
+
+    return G
```
`raise NotImplemented`

mbattifarano commented on src/inverse_astar_search/pastar.py:
```diff
@@ -0,0 +1,157 @@
+
+    # show_network(g0)
+
+    pass
```
remove

mbattifarano commented on test/test_LP1.py:
```diff
@@ -0,0 +1,119 @@
+# from inverse_astar_search.optim import linear_program, get_by_node_pair, entropy
```
remove file

mbattifarano commented on test/test_logit.py:
```diff
@@ -0,0 +1,39 @@
+import hypothesis
```
remove file

pabloguarda replied:
Tests need to be implemented for the methods included in logit.py. I just removed the content of the script but keep the file for further reference.

mbattifarano commented on test/test_pastar.py:
```diff
@@ -0,0 +1,88 @@
+from inverse_astar_search.pastar import max_admissible_heuristic, heuristic_bounds
+
+
+def test_network_lecture_AI():
```
put this in `conftest.py` as `network_lecture_AI`

mbattifarano replied:
`params=[0, 1]` in call to `fixture`
```
@pytest.fixture(params=[0, 1])
def network_lecture_AI(request):
    [...]
    return networks[request.param]
```

mbattifarano commented on test/test_pastar.py:
```diff
@@ -0,0 +1,88 @@
+    return networks
+
+
+def test_astar_admissibility():
```
`def test_astar_admissibility(network_lecture_AI)`

mbattifarano commented on test/test_pastar.py:
```diff
@@ -0,0 +1,88 @@
+    assert optimal_path == astar_optimal_path
+
+
+def test_heuristic_bounds():
```
remove

Commit:
Solving https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418265099

Commit:
Solving https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418266820

Commit:
Solving https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418267047

Commit:
Solving https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418270973

Commit:
Solved https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418271171

Commit:
Solved https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418271717

Commit:
Change based on https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418271932

Commit:
Solved https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418276814

Commit:
Solved https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418278701

Commit:
Changed according to https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418276416

and

https://github.com/mbattifarano/inverse-astar-search/pull/2#discussion_r418278701
Signed-off-by: Pablo <pabloguarda@cmu.edu>

Commit:
Removed unuseful test

Signed-off-by: Pablo <pabloguarda@cmu.edu>

pabloguarda commented:
I am done and ready to merge our pulls to master.
